streams/src/test/java/org/apache/kafka/test/ProcessorTopologyTestDriver.java

```diff
@@ -187,7 +188,7 @@ public ProcessorTopologyTestDriver(StreamsConfig config, TopologyBuilder builder
 
         // Set up all of the topic+partition information and subscribe the consumer to each ...
         for (String topic : topology.sourceTopics()) {
-            TopicPartition tp = new TopicPartition(topic, 1);
+            TopicPartition tp = new TopicPartition(topic, id.partition);
```

Comment:
Should this not be simple `0` -- I guess this should be a proper fix for one bug.

Removing parameter `stateStores` is not a fix (but a nice improvement, so I like it)